python/cudf/cudf/core/dataframe.py

```diff
@@ -2642,9 +2643,8 @@ def merge(
             lhs, rhs, left_on, right_on, how
         )
         # Compute merge
-        gdf_result = libcudf.join.join(
-            lhs._data, rhs._data, left_on, right_on, how, method
-        )
+        gdf_result = join(lhs, rhs, left_on, right_on, how, method)
+        gdf_result = list({v: k for k, v in gdf_result._data.items()}.items())
```

Comment:
Why does the dictionary need to be inverted here? Since it's being converted into a list of tuples, would it suffice to just change the indexing below like `gdf_result[i][1]` to `gdf_result[i][0]`, or does the order of the elements matter somewhere else?

Reply:
There's no real reason besides the legacy code returning with the keys and values switched around, I inverted it just to reduce the degrees of freedom for debugging. Should be removed in the final version.